Geometry_Engine/Compute/CollapseToScaledPolyline.cs

```diff
@@ -41,7 +41,7 @@ public static partial class Compute
         [Input("maxDivisionsPerSegment", "The maximum number of segment each sub-curve can be broken into")]
         [Input("scale", "the radius of a baseline circle for the angle tolerance, smaller circles will have less segments and bigger more. Default of 0 means the program will determine an appropriate radius for the PolyCurve")]
         [Output("C", "A polyline approximating the provided curve")]
-        public static Polyline CollapseToScaledPolyline(this PolyCurve curve, double tolerance = 0.04, double toleranceGrowth = 0.001, int maxDivisionsPerSegment = 200, double scale = 0)
+        public static Polyline CollapseToScaledPolyline(this PolyCurve curve, double tolerance, double toleranceGrowth = 0.001, int maxDivisionsPerSegment = 200, double scale = 0)
```

Comment:
The description for tolerance says that it is an angle tolerance, so is there a reason why it cannot default to `BH.oM.Geometry.Tolerance.Angle`?

Also, the `toleranceGrowth` is equal to `BH.oM.Geometry.Tolerance.MacroDistance` so it might be worthwhile using that one rather than a decimal here to be consistent with other methods/standards? :smile:

Reply:
From memory I only have bad experiences with values under 0.035 or something like that, and would rater let the user make the mistake than have it as default.

`toleranceGrowth` is not a tolerance, but a growth of a tolerance and I figured I should not muddy the waters in this PR by defaulting non-tolerances as tolerances

Reply:
Ok I'm happy with the defence of that then and happy to change my review :smile:

Reply:
I do not know the application of this value, but if `tolerance` argument in this method works differently from what we usually understand by geometrical tolerance in BHoM, maybe we could rename it to something else?

Reply:
I'm going by memory and my old description here but it's an attempt for a tolerance which would equate to the angleTolerance for a circle if the `scale = 0` i.e. auto. otherwise maintain that level of fidelity for curves with more or less curvature without adding an obscene amount of control-points. 

The point of which was to get a good polyline representation of sections without a hundred control points for both, say a circular concrete column and its re-bar holes, as the small holes have no reason for that level of fidelity.

So it's kind of a angleTolerance but not really, as it will be a different angleTolerance based on the other parameters.

So unsure of what would be a good name, which is how I landed on `tolerance` in the first place

Reply:
Hmm not sure. Maybe we could rename the method so that it was pointing directly to integration (it seems to be very specific), e.g. `PrepareToIntegrate` or similar? Then the value could `fidelity`?

Or a step further: make the method `private` and put in integration file? Not sure if that brings any value to the user without the code base that sits on top of it? @IsakNaslundBh do you have opinion on that?